recipes/packio/all/conanfile.py

```diff
@@ -27,81 +33,80 @@ class PackioConan(ConanFile):
     }
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "17"
 
     @property
     def _compilers_minimum_version(self):
-        if scm.Version(self.version) < "2.4.0":
+        if Version(self.version) < "2.4.0":
             return {
-                "apple-clang": 10,
-                "clang": 6,
-                "gcc": 7,
-                "Visual Studio": 16,
+                "apple-clang": "10",
+                "clang": "6",
+                "gcc": "7",
+                "Visual Studio": "16",
+                "msvc": "192",
             }
         return {
-            "apple-clang": 13,
-            "clang": 11,
-            "gcc": 9,
-            "Visual Studio": 16,
+            "apple-clang": "13",
+            "clang": "11",
+            "gcc": "9",
+            "Visual Studio": "16",
+            "msvc": "192",
         }
 
-    def config_options(self):
-        if scm.Version(self.version) < "1.2.0":
-            del self.options.standalone_asio
-        if scm.Version(self.version) < "2.0.0":
-            del self.options.msgpack
-            del self.options.nlohmann_json
-        if scm.Version(self.version) < "2.1.0":
-            del self.options.boost_json
-
-    def requirements(self):
-        if self.options.get_safe("msgpack") or scm.Version(self.version) < "2.0.0":
-            self.requires("msgpack/3.2.1")
+    def export_sources(self):
+        export_conandata_patches(self)
 
-        if self.options.get_safe("nlohmann_json"):
-            self.requires("nlohmann_json/3.9.1")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-         # defaults to True if using boost.asio, False if using asio
-        if self.options.get_safe("boost_json") == "default":
-            self.options.boost_json = not self.options.standalone_asio
+    def package_id(self):
+        self.info.clear()
 
-        if self.options.get_safe("boost_json") or not self.options.get_safe("standalone_asio"):
-            self.requires("boost/1.75.0")
+    @property
+    def _use_boost_json(self):
+        # defaults to True if using boost.asio, False if using asio
+        if self.options.boost_json == "default":
+            return not self.options.standalone_asio
+        return self.options.boost_json
 
-        if self.options.get_safe("standalone_asio"):
-            self.requires("asio/1.18.1")
+    def requirements(self):
+        if self.options.msgpack:
+            self.requires("msgpack-cxx/6.1.1")
+        if self.options.nlohmann_json:
+            self.requires("nlohmann_json/3.11.3")
+        if self._use_boost_json or not self.options.standalone_asio:
+            self.requires("boost/1.83.0")
```

Comment:
Are newer versions not compatible?

Reply:
It avoids a conflict with `boost/1.83.0` in `msgpack-cxx`. It built fine with Boost 1.85.0 otherwise.